osu.Game/Screens/Play/Player.cs

```diff
@@ -267,12 +267,7 @@ private void load(AudioManager audio, OsuConfigManager config, OsuGameBase game,
                 },
                 FailOverlay = new FailOverlay
                 {
-                    SaveReplay = () =>
-                    {
-                        Score.ScoreInfo.Passed = false;
-                        Score.ScoreInfo.Rank = ScoreRank.F;
-                        return prepareAndImportScore();
-                    },
```

Comment:
The logic in here is no longer necessary since failed scores will already be marked with `ScoreRank.F` in either `onFailComplete` or `OnExiting` (in case someone exited between the fail animation).

This shouldn't have exited in the first place, but the reason it did was because previously `ScoreProcessor.Rank.Value` wouldn't update to F rank, and would overwrite the score with a D rank as more hitobjects get judged as "missed" during the fail animation (due to `NewJudgement += _ => ScoreProcessor.PopulateScore(Score)`)